src/utils/v4_position_planner.rs

```diff
@@ -0,0 +1,131 @@
+use super::{
+    to_address,
+    v4_planner::{Actions, V4Planner},
+};
+use crate::{
+    abi::{
+        BurnPositionParams, DecreaseLiquidityParams, IncreaseLiquidityParams, MintPositionParams,
+        SettlePairParams, SweepParams, TakePairParams,
+    },
+    entities::Pool,
+};
+use alloy_primitives::{Address, Bytes, U256};
+use derive_more::derive::{Deref, DerefMut};
+use uniswap_sdk_core::prelude::BaseCurrency;
+use uniswap_v3_sdk::prelude::TickIndex;
+
+#[derive(Clone, Debug, Default, PartialEq, Deref, DerefMut)]
+pub struct V4PositionPlanner(V4Planner);
+
+impl V4PositionPlanner {
+    #[allow(clippy::too_many_arguments)]
+    #[inline]
+    pub fn add_mint<I: TickIndex>(
+        &mut self,
+        pool: &Pool,
+        tick_lower: I,
+        tick_upper: I,
+        liquidity: U256,
+        amount0_max: u128,
+        amount1_max: u128,
+        owner: Address,
+        hook_data: Bytes,
+    ) {
+        self.add_action(&Actions::MINT_POSITION(MintPositionParams {
+            poolKey: pool.pool_key.clone(),
+            tickLower: tick_lower.to_i24(),
+            tickUpper: tick_upper.to_i24(),
+            liquidity,
+            amount0Max: amount0_max,
+            amount1Max: amount1_max,
+            owner,
+            hookData: hook_data,
+        }));
+    }
+
+    #[inline]
+    pub fn add_increase(
+        &mut self,
+        token_id: U256,
+        liquidity: U256,
+        amount0_max: u128,
+        amount1_max: u128,
+        hook_data: Bytes,
+    ) {
+        self.add_action(&Actions::INCREASE_LIQUIDITY(IncreaseLiquidityParams {
+            tokenId: token_id,
+            liquidity,
+            amount0Max: amount0_max,
+            amount1Max: amount1_max,
+            hookData: hook_data,
+        }));
+    }
+
+    #[inline]
+    pub fn add_decrease(
+        &mut self,
+        token_id: U256,
+        liquidity: U256,
+        amount0_min: u128,
+        amount1_min: u128,
+        hook_data: Bytes,
+    ) {
+        self.add_action(&Actions::DECREASE_LIQUIDITY(DecreaseLiquidityParams {
+            tokenId: token_id,
+            liquidity,
+            amount0Min: amount0_min,
+            amount1Min: amount1_min,
+            hookData: hook_data,
+        }));
+    }
+
+    #[inline]
+    pub fn add_burn(
+        &mut self,
+        token_id: U256,
+        amount0_min: u128,
+        amount1_min: u128,
+        hook_data: Bytes,
+    ) {
+        self.add_action(&Actions::BURN_POSITION(BurnPositionParams {
+            tokenId: token_id,
+            amount0Min: amount0_min,
+            amount1Min: amount1_min,
+            hookData: hook_data,
+        }));
+    }
+
+    #[inline]
+    pub fn add_settle_pair(
+        &mut self,
+        currency0: &impl BaseCurrency,
+        currency1: &impl BaseCurrency,
+    ) {
+        self.0.add_action(&Actions::SETTLE_PAIR(SettlePairParams {
```

Comment:
```suggestion
        self.add_action(&Actions::SETTLE_PAIR(SettlePairParams {
```

Reply:
this actually fails the linter

Reply:
and it's valid

Reply:
Because `derive_more` is not in Cargo.toml.

Reply:
even when it's there, the linter still fails